File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        iter = head
        ln = None
        while iter != None:
            ln = ListNode(val=iter.val, next=ln)
            iter = iter.next
        return ln

File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-0.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        ListNode dummy {0};
        ListNode* t = &dummy;
        while(list1 != nullptr || list2 != nullptr) {
            if (list1 != nullptr && list2 != nullptr) {
                if ( list1->val <= list2->val) {
                    t->next = list1;
                    list1 = list1->next;
                } else {
                    t -> next = list2;
                    list2 = list2->next;
                }
            t = t->next;
            } else {
                if (list1 == nullptr) {
                    t ->next = list2;
                } else if (list2 == nullptr) {
                    t->next = list1;
                }
                break;
            }
        }
        return dummy.next;
    }
};
